src/main/java/de/unistuttgart/iste/gits/media_service/service/MediaService.java

```diff
@@ -97,6 +97,37 @@ public List<MediaRecord> getMediaRecordsById(List<UUID> ids, boolean generateUpl
         );
     }
 
+    /**
+     * The same as {@link #getMediaRecordsByIds(List, boolean, boolean)}, except that it doesn't throw an exception
+     * if an entity cannot be found. Instead, it returns NULL for that entity.
+     *
+     * @return Returns a List containing the MediaRecords with the specified ids. If a media record for an id cannot
+     *         be found, returns NULL for that media record instead.
+     */
+    public List<MediaRecord> findMediaRecordsByIds(List<UUID> ids, boolean generateUploadUrls, boolean generateDownloadUrls) {
+        List<MediaRecordEntity> records = repository.findAllById(ids).stream().toList();
+
+        List<MediaRecord> result = new ArrayList<>(ids.size());
+
+        // go over all requested ids
+        for (UUID id : ids) {
+            // get the entity with the matching id or NULL if it doesn't exist
+            MediaRecordEntity entity = records.stream().filter(x -> x.getId().equals(id)).findAny().orElse(null);
+            MediaRecord mediaRecord = null;
```

Comment:
,just a tip, no todo, you can also use
`findAny().map(entity -> modelMapper.map(entity, MediaRecord.class)).orElse(null);